state/src/replicationv2/network/gossip.rs

```diff
@@ -0,0 +1,86 @@
+//! Gossip networking interface, acts as a shim between raft and our gossip
+//! layer
+
+use gossip_api::request_response::{GossipRequest, GossipResponse};
+use job_types::network_manager::{NetworkManagerJob, NetworkManagerQueue};
+use openraft::{
+    error::{RPCError, RaftError},
+    RaftNetworkFactory,
+};
+
+use crate::replicationv2::{
+    error::{new_network_error, ReplicationV2Error},
+    Node, NodeId, TypeConfig,
+};
+
+use super::{P2PRaftNetwork, P2PRaftNetworkWrapper, RaftRequest, RaftResponse};
+
+/// The error message emitted when a response type is invalid
+const ERR_INVALID_RESPONSE: &str = "invalid response type from raft peer";
+
+/// The network shim
+#[derive(Clone)]
+pub struct GossipNetwork {
```

Comment:
I understand it's a bit out of scope for our codebase, but could we have some quick documentation about how this struct is managed via the `RaftNetworkFactory` trait? IIUC every time there's a membership change, new `GossipNetwork` instances will be spawned for each node in the membership via `RaftNetworkFactory::new_client`. This is initially unclear w/o looking at the openraft guide and I was a bit confused as to why the whole `GossipNetwork` only has one `target`